core/src/bundle/usCoreBundleContext.cpp

```diff
@@ -59,19 +59,19 @@ std::map<std::string, Any> InitProperties(std::map<std::string, Any> configurati
   return configuration;
 }
 
-CoreBundleContext::CoreBundleContext(const std::map<std::string, Any>& props)
+CoreBundleContext::CoreBundleContext(const std::map<std::string, Any>& props, std::ostream* logger)
   : id(globalId++)
+  , frameworkProperties(InitProperties(props))
+  , sink(std::make_shared<LogSink>(logger, any_cast<bool>(frameworkProperties.at(Constants::FRAMEWORK_LOG))))
```

Comment:
I agree, let's create a separate issue for the shared_ptr log sink.

Your reasoning about the exception handling makes sense, let's stick 
with your solution.


On 07/27/2016 09:06 PM, Jeff DiClemente wrote:
>
> In core/src/bundle/usCoreBundleContext.cpp 
> <https://github.com/CppMicroServices/CppMicroServices/pull/96#discussion_r72502926>:
>
> >    : id(globalId++)
> > +  , frameworkProperties(InitProperties(props))
> > +  , sink(std::make_shared<LogSink>(logger, any_cast<bool>(frameworkProperties.at(Constants::FRAMEWORK_LOG))))
>
> I share your skepticism about the use of |shared_ptr|. Can we defer 
> exploring changing it to |unique_ptr| and capture that work in a issue?
>
> Yes, it is intentional to throw if the framework property is not a 
> |bool|. I believe that we should fail early and force the user to do 
> the right thing (i.e. use a |bool|) instead of trying to account for 
> possible user error (e.g. a user using a |string| instead of |bool|). 
> Since this is a Framework defined property I think we should enforce 
> its correct use.
> If the user does get an exception, the exception message plus adequate 
> documentation about the framework properties should be enough for 
> users to correct their mistake. Silently failing and using a default 
> value would cause user confusion IMHO.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CppMicroServices/CppMicroServices/pull/96/files/4a351c96b003299991ca585f0b452d3d1b489e68#r72502926>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AANXpWpC5NriGB1qLgrwA2p3QFoe5paSks5qZ6wggaJpZM4IcUhJ>.
>